motionplan/tpSpaceRRT_test.go

```diff
@@ -0,0 +1,215 @@
+	"go.viam.com/test"
+
+	"go.viam.com/rdk/motionplan/tpspace"
+	"go.viam.com/rdk/referenceframe"
+	"go.viam.com/rdk/spatialmath"
+)
+
+func TestPtgRrt(t *testing.T) {
+	logger := golog.NewTestLogger(t)
+	ackermanFrame, err := NewptgFrame("test")
+	test.That(t, err, test.ShouldBeNil)
+
+	goalPos := spatialmath.NewPose(r3.Vector{X: 50, Y: 10, Z: 0}, &spatialmath.OrientationVectorDegrees{OZ: 1, Theta: 180})
+
+	opt := newBasicPlannerOptions()
+	opt.SetGoalMetric(NewPositionOnlyMetric(goalPos))
+	opt.DistanceFunc = SquaredNormNoOrientSegmentMetric
+	opt.GoalThreshold = 10.
+	mp, err := newTPSpaceMotionPlanner(ackermanFrame, rand.New(rand.NewSource(42)), logger, opt)
+	test.That(t, err, test.ShouldBeNil)
+	tp, ok := mp.(*tpspaceRRTMotionPlanner)
+	test.That(t, ok, test.ShouldBeTrue)
+
+	_, err = tp.plan(context.Background(), goalPos, nil)
+	test.That(t, err, test.ShouldBeNil)
+}
+
+func TestPtgWithObstacle(t *testing.T) {
+	logger := golog.NewTestLogger(t)
+	ackermanFrame, err := NewptgFrame("ackframe")
+	test.That(t, err, test.ShouldBeNil)
+
+	goalPos := spatialmath.NewPoseFromPoint(r3.Vector{X: 5000, Y: 0, Z: 0})
+
+	fs := referenceframe.NewEmptyFrameSystem("test")
+	fs.AddFrame(ackermanFrame, fs.World())
+
+	opt := newBasicPlannerOptions()
+	opt.SetGoalMetric(NewPositionOnlyMetric(goalPos))
+	opt.DistanceFunc = SquaredNormNoOrientSegmentMetric
+	opt.GoalThreshold = 10.
+	// obstacles
+	obstacle1, err := spatialmath.NewBox(spatialmath.NewPoseFromPoint(r3.Vector{2500, -500, 0}), r3.Vector{180, 1800, 1}, "")
+	test.That(t, err, test.ShouldBeNil)
+	obstacle2, err := spatialmath.NewBox(spatialmath.NewPoseFromPoint(r3.Vector{2500, 2000, 0}), r3.Vector{180, 1800, 1}, "")
+	test.That(t, err, test.ShouldBeNil)
+	obstacle3, err := spatialmath.NewBox(spatialmath.NewPoseFromPoint(r3.Vector{2500, -1400, 0}), r3.Vector{50000, 10, 1}, "")
+	test.That(t, err, test.ShouldBeNil)
+	obstacle4, err := spatialmath.NewBox(spatialmath.NewPoseFromPoint(r3.Vector{2500, 2400, 0}), r3.Vector{50000, 10, 1}, "")
+	test.That(t, err, test.ShouldBeNil)
+
+	geoms := []spatialmath.Geometry{obstacle1, obstacle2, obstacle3, obstacle4}
+
+	worldState, err := referenceframe.NewWorldState(
+		[]*referenceframe.GeometriesInFrame{referenceframe.NewGeometriesInFrame(referenceframe.World, geoms)},
+		nil,
+	)
+	test.That(t, err, test.ShouldBeNil)
+	sf, err := newSolverFrame(fs, "ackframe", referenceframe.World, nil)
+	test.That(t, err, test.ShouldBeNil)
+	collisionConstraints, err := createAllCollisionConstraints(sf, fs, worldState, referenceframe.StartPositions(fs), nil)
+	test.That(t, err, test.ShouldBeNil)
+
+	for name, constraint := range collisionConstraints {
+		opt.AddStateConstraint(name, constraint)
+	}
+
+	mp, err := newTPSpaceMotionPlanner(ackermanFrame, rand.New(rand.NewSource(42)), logger, opt)
+	test.That(t, err, test.ShouldBeNil)
+	tp, _ := mp.(*tpspaceRRTMotionPlanner)
+
+	_, err = tp.plan(context.Background(), goalPos, nil)
+	test.That(t, err, test.ShouldBeNil)
```

Comment:
Same thing here, except that the plan should probably output more than 2 waypoints